source/slang/slang-ir-layout.cpp

```diff
@@ -692,6 +694,53 @@ struct Std140LayoutRules : IRTypeLayoutRules
     }
 };
 
+struct LLVMLayoutRules : IRTypeLayoutRules
```

Comment:
It feels a little weird to have "LLVM" be treated like a target, when its really more of a pathway for generating code on one or more targets. I'm not sure there's anything to do about it right now, but it is notable that both the GLSL and SPIR-V back-ends for Vulkan utilize the exact same layout implementations, because the layout rules are a property of the "final" target platform (Vulkan) and not of the means by which we access it.

Again, I'm not sure what to do about it right now, but it seems that if we start using LLVM to generate code for more target platforms over time (which seems likely), we'd need a clear mental model for how that should interact with things like layout rules for those target platforms.

Reply:
The thinking I had with `LLVMLayoutRules` is that it would query and act according to the native data layout rules for the specified LLVM target triple. So at least here, the mental model isn't exactly that LLVM would be a specific target, but that it's a class of targets with the same implementation for querying the layout rules. Although, right now it doesn't query the rules from LLVM; that's left as future work currently, as implementing it seems to require some larger refactoring around layout rules that's probably better to do in a separate PR.

Still, this PR does currently conflate LLVM with being the "CPU" target quite heavily, and that aspect should be improved. On `-emit-cpu-code-via-llvm`, I think it would be a good idea. (more on this in the `slang-target.h` comment below).